lib/open_api_spex/schema.ex

```diff
@@ -432,28 +432,42 @@ defmodule OpenApiSpex.Schema do
   def validate(%Schema{type: type}, nil, path, _schemas) do
     {:error, "#{path}: null value where #{type} expected"}
   end
-  def validate(schema = %Schema{type: type}, value, path, _schemas) when type in [:integer, :number] do
+  def validate(schema = %Schema{type: :integer}, value, path, _schemas) when is_integer(value) do
+    # TODO DUPLICATION
```

Comment:
Personally not a fan of TODO comments. Looks like the duplication here is OK, or a shared implementation could be factored out.